src/PopEngine/index.ts

```diff
@@ -89,18 +112,27 @@ export class PopEngine {
   public showPop(targetElement: Element, isPinned: boolean, pop: Pop): void {
```

Comment:
As we've already noted, but we should definitely come back to this with new ways to tackle this function since its super overloaded.